examples/flutter_gallery/lib/gallery/app.dart

```diff
@@ -157,21 +157,27 @@ class GalleryAppState extends State<GalleryApp> {
       };
     }
 
-    return new MaterialApp(
-      title: 'Flutter Gallery',
-      color: Colors.grey,
-      theme: _galleryTheme.theme.copyWith(platform: _platform ?? defaultTargetPlatform),
-      showPerformanceOverlay: _showPerformanceOverlay,
-      checkerboardRasterCacheImages: _checkerboardRasterCacheImages,
-      checkerboardOffscreenLayers: _checkerboardOffscreenLayers,
-      routes: _kRoutes,
-      home: home,
-      builder: (BuildContext context, Widget child) {
-        return new Directionality(
-          textDirection: _overrideDirection,
-          child: _applyScaleFactor(child),
-        );
-      },
+    return new Semantics(
```

Comment:
Is this required for every App? Should it be part of MaterialApp? What changes if I don't specify a top level textDirection for semantics?

Reply:
I guess it reverts to paint order?

Reply:
Reverting for now. Let's address this in https://github.com/flutter/flutter/issues/16835.